moped-editor/src/views/projects/projectView/ProjectComponents/utils/useComponentLinkParams.js

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useState } from "react";
+
+/** Update search params without the router triggering a re-render
+ * @see https://github.com/remix-run/react-router/discussions/9851#discussioncomment-5934435
+ * @see https://github.com/kentcdodds/kentcdodds.com/blob/027070cdb2742452f98011c114a03ee325052cc7/app/utils/misc.tsx#L298-L325
+ * Comment below is adapted from the linked code to apply to our use case in the project components map
+ */
+const updateParamsWithoutRender = (queryKey, queryValue) => {
```

Comment:
so glad you got this work! this looks like the pro version of what I set up [on the lookups view](https://github.com/cityofaustin/atd-moped/blob/ee9e54e91283397bcef194b93dc8986eb13bf1f7/moped-editor/src/views/dev/LookupsView/index.js#L58-L63). that iteration is much simpler bc we don't have to worry about multiple params in the URL. this looks really good! 🔥